apps/persons/tests/test_person_plugin.py

```diff
@@ -0,0 +1,88 @@
+"""
+Unit tests for the Person plugin and its model
+"""
+from django.core.files.base import ContentFile
+from django.template import RequestContext
+from django.test import TestCase
+
+from filer.models import Image
+
+from cms.api import create_page, add_plugin
+from cms.plugin_rendering import ContentRenderer
+
+from djangocms_picture.cms_plugins import PicturePlugin
+from djangocms_text_ckeditor.cms_plugins import TextPlugin
+
+from apps.core.factories import UserFactory
+
+from ..cms_plugins import PersonPlugin
+from ..factories import PersonFactory
+
+
+class PersonPluginTestCase(TestCase):
+    """
+    Test that PersonPlugin correctly displays a Person's page placeholders content
+    """
+
+    def test_person_plugin_render(self):
+        """
+        Test that a PersonPlugin correctly renders person's page specific information
+        """
+        # Create a filer fake image
+        user = UserFactory(is_staff=True, is_superuser=True)
+        file_obj = ContentFile(b"portrait image")
+        image = Image.objects.create(
+            owner=user, original_filename="portrait.png", file=file_obj
+        )
+        # Create a Person
+        person = PersonFactory()
+        person_page = person.extended_object
+        # Add portrait to related placeholder
+        placeholder = person_page.placeholders.get(slot="portrait")
+        add_plugin(
+            placeholder,
+            PicturePlugin,
+            "en",
+            **{"picture": image, "attributes": {"alt": "portrait alternate"}}
+        )
+        # A resume to related placeholder
+        placeholder = person_page.placeholders.get(slot="resume")
+        add_plugin(
+            placeholder,
+            # We use a HTML plugin as Django-CMS do not have pure text plugin by default
+            TextPlugin,
+            "en",
+            **{"body": "A short resume"}
+        )
+        # Create a page to add the plugin to
+        page = create_page("A page", template="richie/fullwidth.html", language="en")
+        # Add the PersonPlugin to page's main placeholder
+        placeholder = page.placeholders.get(slot="maincontent")
+        person_plugin = add_plugin(
+            placeholder, PersonPlugin, "en", **{"person": person}
+        )
+
+        # Render the PersonPlugin
+        request = {}
+        renderer = ContentRenderer(request)
+        context = RequestContext(request)
+        context["request"] = request
+        html = renderer.render_plugin(person_plugin, context)
+
+        # Page title should be present
+        self.assertIn(
+            '<a href="{url}">'.format(url=person_page.get_absolute_url()), html
+        )
+        # Person's portrait and its properties should be present
+        self.assertIn('<img src="" alt="{alt}">'.format(alt="portrait alternate"), html)
```

Comment:
src should not be equal to a blanck string

Reply:
It's a fake image. How should we implement a test of this kind without storing any file ?